<?xml version="1.0"?>

<launch>
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

  <include file="$(find bebop_gazebo)/launch/spawn_bebop.launch" />

  <node name="spawn_helipad" pkg="gazebo_ros" type="spawn_model"
      args="-file $(find bebop_gazebo)/models/marker3/model.sdf
         -model marker3 -sdf
         -x 10.0
         -y -0.0
         -z 0.3"
      respawn="false" output="screen"/>

  <node pkg="bebop_gazebo" type="moving_helipad.py" name="moving_helipad" output="screen" >
      <param name="x_vel" type="double" value="-2.0" />
      <param name="y_vel" type="double" value="0.0" />
      <param name="seconds_before_moving" type="double" value="10" />
  </node>

  <node pkg="rostopic" type="rostopic" name="takeoffbaby" args="pub /action/takeoff/goal hector_uav_msgs/TakeoffActionGoal '{header: { seq: 0, stamp: { secs: 0, nsecs: 0 }, frame_id: '' }, goal_id: {stamp: { secs: 0, nsecs: 0}, id: ''}, goal: {} }'" output="screen"/>

</launch>